src/dune_engine/scheduler.mli

```diff
@@ -138,3 +138,7 @@ val running_jobs_count : t -> int
 val shutdown : unit -> unit Fiber.t
 
 val inject_memo_invalidation : Memo.Invalidation.t -> unit Fiber.t
+
+(** [sleep duration] wait for [duration] to elapse. If successful, [`Finished]
+    is returned. Otherwise, the timer is [`Cancelled] *)
+val sleep : float -> [ `Finished | `Cancelled ] Fiber.t
```

Comment:
I find it surprising that `sleep` may return `Cancelled` but the API provides no way to cancel a sleep.

Reply:
Indeed that's a bit surprising. The current behavior is that all timers are cancelled after `Scheduler.go` if they aren't waited for. So the use can still observe the cancellation. I could change this behavior to raise instead.

Reply:
Simplified the return value to `unit`.